shared/src/main/java/org/odk/collect/shared/locks/ChangeLock.kt

```diff
@@ -25,6 +25,6 @@ interface ChangeLock {
     fun unlock(token: Any)
 
     companion object {
-        val defaultToken = Any()
```

Comment:
I think this still makes sense to be an `Any` (for the reasons outlined in https://github.com/getodk/collect/pull/6747#pullrequestreview-2880959043), but all the infrastructure for that is in place so need to block the PR!